Tests/SymbolKitTests/SymbolGraph/Symbol/ValueConstraintsTests.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+import SymbolKit
+
+class ValueConstraintsTests: XCTestCase {
+    func testValueConstraintsCanBeDecoded() throws {
+        let jsonData = """
+          {
+            "accessLevel" : "public",
+            "identifier" : {
+              "interfaceLanguage" : "data",
+              "precise" : "data:example:Dictionary@property1"
+            },
+            "kind" : {
+              "displayName" : "Dictionary Key",
+              "identifier" : "dictionaryKey"
+            },
+            "names" : {
+              "title" : "property1"
+            },
+            "pathComponents": [],
+            "minimum": 3,
+            "maximum": 3.5,
+            "default": "str",
+            "allowedValues": ["a", 1, null],
+        }
+        """.data(using: .utf8)
+        
+        let decoder = JSONDecoder()
+        let symbol = try decoder.decode(SymbolGraph.Symbol.self, from: jsonData!)
+        
+        if let minimum = symbol.mixins[SymbolGraph.Symbol.Minimum.mixinKey] as? SymbolGraph.Symbol.Minimum {
+            if case let .integer(value) = minimum.value {
+                XCTAssertEqual(value, 3)
+            } else {
+                XCTFail("Unable to extract integer value from minimum JSON.")
+            }
```

Comment:
Also, since `AnyScalar` and `AnyNumber` are equatable, this could also be written as:

```swift
XCTAssertEqual(minimum.value, .integer(3))
```